lib/Runtime/Types/TypePropertyCache.cpp

```diff
@@ -172,7 +172,9 @@ namespace Js
                 &propertyIndex,
                 &isInlineSlot,
                 &isMissing,
-                &prototypeObjectWithProperty))
+                &prototypeObjectWithProperty)
+            || (prototypeObjectWithProperty != nullptr
+                && prototypeObjectWithProperty != propertyObject))
```

Comment:
Yes, as I thought, the byte code gen for super.x is misusing the inline caches. So let's apply a fix there rather than in the cache logic. We can discuss offline.